### PR TITLE
Add check of application of overrides to the crudtester

### PR DESCRIPTION
**Pull request by marun**
Previously there was no testing of overrides being applied in integration and e2e testing.  As part of this PR, the marshaling of overrides was refactored for reuse.

Targets #255

### CODE REVIEW

**Comment by marun** [comment]
Adding wip - intend to unify marshaling between the sync controller and crudtester.

**Comment by marun** [comment]
Rebased

File: pkg/controller/util/overrides.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	return &ClusterOverrides{
+		Path:      path,
+		Overrides: overrides,
+	}, nil
+}
+
+func marshallOverrides(typeConfig typeconfig.Interface, override *unstructured.Unstructured) (map[string]interface{}, []string, error) {
+	overrideMap := make(map[string]interface{})
+	overridePath := []string{}
+	if typeConfig.GetOverride() != nil && override != nil {
```
Comment (font) [inline pkg/controller/util/overrides.go]: (no action required) Consider performing the nil check first and returning immediately.

File: pkg/controller/util/overrides.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return &ClusterOverrides{
+		Path:      path,
+		Overrides: overrides,
+	}, nil
+}
+
+func marshallOverrides(typeConfig typeconfig.Interface, override *unstructured.Unstructured) (map[string]interface{}, []string, error) {
```
Comment (font) [inline pkg/controller/util/overrides.go]: (no action required) is marshall the right term here? It seems like we're deserializing the unstructured data into a meaningful map and string slice.

**Pull request merged**